consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java

```
@@ -63,23 +64,24 @@ public static ProtocolSchedule create(
       final BadBlockManager badBlockManager,
       final boolean isParallelTxProcessingEnabled,
       final boolean isBlockAccessListEnabled,
-      final MetricsSystem metricsSystem) {
+      final MetricsSystem metricsSystem,
+      final EvmConfiguration evmConfiguration) {
 
     Map<Long, Function<ProtocolSpecBuilder, ProtocolSpecBuilder>> postMergeModifications =
         new HashMap<>();
     postMergeModifications.put(
         0L,
         (specBuilder) ->
             MergeProtocolSchedule.applyParisSpecificModifications(
-                specBuilder, config.getChainId()));
+                specBuilder, config.getChainId(), evmConfiguration));
     unapplyModificationsFromShanghaiOnwards(config, postMergeModifications);
 
     return new ProtocolScheduleBuilder(
             config,
             Optional.of(DEFAULT_CHAIN_ID),
             new ProtocolSpecAdapters(postMergeModifications),
             isRevertReasonEnabled,
-            EvmConfiguration.DEFAULT,
+            evmConfiguration,
```

Wondering what the impact of this is - I think is getting overridden by the forks milestones that are later added anyway?

There are others similar places such as QbftProtocolScheduleBuilder that you've left untouched, but the flag activation appears to work as expected despite this.

---

So `QbftBesuControllerBuilder` inherits the `EvmConfiguration` from its parent `BesuControllerBuilder` so it will always be in place from what I can see but please double check. There is a method in `QbftBesuControllerBuilder:140` that forces it to `EvmConfiguration.DEFAULT` but is unused, maybe it should be removed to not mislead anyone?

I did check and all the places where `EvmConfiguration.DEFAULT` is used seem to be only tests.

---

>There is a method in QbftBesuControllerBuilder:140 that forces it to EvmConfiguration.DEFAULT but is unused, maybe it should be removed to not mislead anyone?

I will remove this one, git blame shows it was only added for `evmtool` support.